src/fan.mli

```diff
@@ -22,9 +22,11 @@ type 'a t
 val equal : 'a t -> 'a t -> bool
 (** The equality function for fanouts. *)
 
-val v : hash_size:int -> entry_size:int -> int -> [ `Write ] t
-(** [v ~hash_size ~entry_size n] creates a fan_out for an index with [hash_size]
-    and [entry_size], containing [n] elements. *)
+val v : hash_size:int -> entry_size:int -> int -> int option -> [ `Write ] t
+(** [v ~hash_size ~entry_size n os] creates a fan_out for an index with
+    [hash_size] and [entry_size], containing [n] elements. If the fan is created
+    while another one was already present, a warning is issued if the new fan is
+    twice the previous one *)
```

Comment:
The fan-out implementation itself is agnostic to the existence of a previous fan-out, so I think this logging should be done externally. i.e. build the new fan out, query `Fan.nb_fans`, then log the size difference as part of the merge logic.

Reply:
(API and comment change  now need to go too.)